### ISSUE TITLE
Password change plugin adding empty passwords

### ISSUE THREAD

**Issue by fluteze** [roundcube/roundcubemail#7723]
The password change plugin returns empty an password field if the "$config['password_algorithm']"  contains an invalid entry.
This is a result of the switch/case statement having no default action. Here is a patch to fix this issue:
```
`--- password.php.orig	2020-11-14 12:26:00.748848486 -0700
+++ password.php	2020-11-14 12:24:07.090332419 -0700
@@ -768,6 +778,11 @@
         case 'clear':
             $crypted = $password;
 	    break;
+
+	default:
+            return false;
+            break;
+
`

**Comment by johndoh** [comment]
A simpler fix might be to change [password.php line 523](https://github.com/roundcube/roundcubemail/blob/master/plugins/password/password.php#L523) to `$crypted = null;` as this is already caught on line 774.

Please could you submit a PR?

**Comment by fluteze** [comment]
I looked at $crypted = null; but that doesn't appear to catch typos or unimplemented methods. I'd prefer returning an error as a better option than silently returning an unencrypted string when encryption is expected. Thoughts?

**Comment by johndoh** [comment]
Nothing against using the default option it just looked to me like `$crypt` should be initialized as null rather than an empty string and with that there is no need for it.

**Comment by alecpl** [comment]
On invalid/unsupported value I would just throw a hard error.

**Comment by fluteze** [comment]
Merged in PR #7735
